admin/services/group.service.js

```diff
@@ -1,49 +1,116 @@
+  // return groupDataService.getGroups(query)
+  return groupDataService.sqlGetGroups()
 }
 
+/**
+ * Get pupils filtered by schoolId and groupId.
+ * @param schoolId
+ * @param groupIdToExclude
+ * @returns {Promise<*>}
+ */
 groupService.getPupils = async function (schoolId, groupIdToExclude) {
   if (!schoolId) { return false }
-  let query = {}
-  let filteredPupil = []
-  let groupedPupils = []
-  let pupils = await pupilDataService.getSortedPupils(schoolId)
-
-  if (groupIdToExclude) {
-    query = { '_id': { $ne: groupIdToExclude } }
-  }
-  const groups = await groupDataService.getGroups(query)
-  groups.map((group) => {
-    const pupils = Object.values(group.pupils)
-    pupils.forEach(id => {
-      return groupedPupils.push(id)
-    })
-  })
+  return groupDataService.sqlGetPupils(schoolId, groupIdToExclude)
 
-  pupils.map((pupil) => {
-    if (!groupedPupils.includes(pupil._id.toString())) {
-      filteredPupil.push(pupil)
-    }
-  })
-
-  return filteredPupil
+  // @TODO: TO BE DELETED
+  // if (!schoolId) { return false }
+  // let query = {}
+  // let filteredPupil = []
+  // let groupedPupils = []
+  // let pupils = await pupilDataService.getSortedPupils(schoolId)
+  //
+  // if (groupIdToExclude) {
+  //   query = { '_id': { $ne: groupIdToExclude } }
+  // }
+  // const groups = await groupDataService.getGroups(query)
+  // groups.map((group) => {
+  //   const pupils = Object.values(group.pupils)
+  //   pupils.forEach(id => {
+  //     return groupedPupils.push(id)
+  //   })
+  // })
+  //
+  // pupils.map((pupil) => {
+  //   if (!groupedPupils.includes(pupil._id.toString())) {
+  //     filteredPupil.push(pupil)
+  //   }
+  // })
+  //
+  // return filteredPupil
 }
 
+/**
+ * Get group by id.
+ * @param groupId
+ * @returns {Promise<*>}
+ */
 groupService.getGroupById = async function (groupId) {
   if (!groupId) { return false }
-  return groupDataService.getGroup({'_id': groupId})
+  // @TODO: TO BE DELETED
+  // return groupDataService.getGroup({'_id': groupId})
+  return groupDataService.sqlGetGroup(groupId)
+}
+
+/**
+ * Update group (group and pupils assigned to groups).
+ * @param id
+ * @param group
+ * @returns {Promise<boolean>}
+ */
+groupService.updateGroup = async (id, group) => {
+  if (!id || !group || !group.name) { return false }
+  return new Promise(async (resolve, reject) => {
+    try {
+      await groupDataService.sqlUpdate(id, group.name)
+      await groupDataService.sqlAssignPupilsToGroup(id, group.pupils)
+      resolve(group)
+    } catch (error) {
+      reject(error)
+    }
+  })
 }
 
-groupService.getGroupByName = async function (groupName) {
-  if (!groupName) { return false }
-  return groupDataService.getGroup({'name': groupName})
+/**
+ * Create group.
+ * @param group
+ * @param groupPupils
+ * @returns {Promise<boolean>}
+ */
+groupService.create = async (group, groupPupils) => {
+  if (!group) { return false }
+  return new Promise(async (resolve, reject) => {
```

Comment:
Promise not needed in an async function.  Please remove and use throw to reject and return to resolve.